File: un6unw/Week32/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int validCount = 0;

        for (String tree : skill_trees) {
            StringBuilder filtered = new StringBuilder();

            // 스킬트리에서 선행 스킬만 골라냄
            for (char c : tree.toCharArray()) {
                if (skill.indexOf(c) != -1) {
                    filtered.append(c);
                }
            }

            // skill의 접두사(prefix)인지 확인
            if (skill.startsWith(filtered.toString())) {
                validCount++;
            }
        }

        return validCount;
    }
}

File: ity0526/ity0562_Week32/스킬트리.py
def solution(skill, skill_trees):
    answer = len(skill_trees)
    s_index = 0

    for skill_tree in skill_trees:
        for s in skill_tree:
            if s in skill:
                if s == skill[s_index]:
                    s_index += 1
                else:
                    answer -= 1
                    break

        s_index = 0

    return answer